src/TSTransformer/util/jsx/getJsxIndex.ts

```diff
@@ -1,10 +1,11 @@
 import luau from "@roblox-ts/luau-ast";
 import assert from "assert";
 import { TransformState } from "TSTransformer/classes/TransformState";
+import { propertyAccessExpressionChain } from "TSTransformer/util/expressionChain";
 
 function getJsxIndex(factory: string) {
-	const [namespace, index] = factory.split(".");
-	return luau.property(luau.id(namespace), index);
+	const [namespace, ...path] = factory.split(".");
+	return propertyAccessExpressionChain(luau.id(namespace), path);
```

Comment:
What's the behaviour for this if the factory is just `""`? I think it'll generate an empty luau.id, so the emit would become something like `local element = ("className")`, as opposed to `local element = factory("className")`

Reply:
It might also be wrong if the factory is set to something like `123.createJsx`?

Reply:
<img width="960" alt="image" src="https://github.com/roblox-ts/roblox-ts/assets/9200592/13c51940-68e8-464a-b89f-395ebf8db470">

Looks like non-property chains aren't allowed?

Reply:
Weird.. I'm not getting that error in Visual Studio Code, but it does throw a diagnostic when compiling.
What about Luau identifier rules? Iirc some TS variable names wouldn't be valid in Luau.

Reply:
Side stepped this issue through the new "entity name" stuff in latest commits. It will receives TS expressions and then just compiles them like anything else might be. So we'd get the same sort of identifier diagnostics.

Reply:
EntityName goes to `transformIdentifierDefined` though, which itself sidesteps the regular identifier handling + diagnostics. This means the valid identifier check is never done, as well as things like hoisting (which I believe would technically be possible for a jsx factory)